modules/miner/mine.go

```diff
@@ -85,24 +84,27 @@ func (m *Miner) submitBlock(b types.Block) error {
 // time to complete, and should not be called with a lock.
 func (m *Miner) solveBlock(blockForWork types.Block, blockMerkleRoot crypto.Hash, target types.Target) (b types.Block, solved bool, err error) {
 	b = blockForWork
-	hashbytes := make([]byte, 72)
+	hashbytes := make([]byte, 80)
 	copy(hashbytes, b.ParentID[:])
-	copy(hashbytes[40:], blockMerkleRoot[:])
-
-	nonce := (*uint64)(unsafe.Pointer(&hashbytes[32]))
-	*nonce = b.Nonce
-	for i := 0; i < iterationsPerAttempt; i++ {
-		id := crypto.HashBytes(hashbytes)
-		if bytes.Compare(target[:], id[:]) >= 0 {
-			b.Nonce = binary.LittleEndian.Uint64(hashbytes[32:40])
-			err = m.submitBlock(b)
-			if err != nil {
+	binary.LittleEndian.PutUint64(hashbytes[40:48], uint64(b.Timestamp))
+	copy(hashbytes[48:], blockMerkleRoot[:])
```

Comment:
I thought it would be simpler / more clear to do it this way. I have no idea which is faster, but my loose experimentation suggests that it really doesn't make a difference.